jme3-effects/src/main/java/com/jme3/water/WaterFilter.java

```diff
@@ -382,6 +405,14 @@ public void read(JmeImporter im) throws IOException {
 
         useCaustics = ic.readBoolean("useCaustics", true);
 
+        causticsTexture = (Texture2D) ic.readSavable("causticsTexture", null);
+        heightTexture = (Texture2D) ic.readSavable("heightTexture", null);
+        normalTexture = (Texture2D) ic.readSavable("normalTexture", null);
+        foamTexture = (Texture2D) ic.readSavable("foamTexture", null);
+        needSaveReflectionScene = ic.readBoolean("needSaveReflectionScene", false);
+        reflectionScene = (Spatial) ic.readSavable("reflectionScene", null);
```

Comment:
Read the texture path, load the texture if it's relevant, else keep the texture null.